File: AdminPanel/AdminPanel.Web/Modules/Souqcom/Products/ProductsColumns.cs
namespace AdminPanel.Souqcom.Columns;

[ColumnsScript("Souqcom.Products")]
[BasedOnRow(typeof(ProductsRow), CheckNames = true)]
public class ProductsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Discription { get; set; }
    public decimal Price { get; set; }
    public string CatidName { get; set; }
    public string Photo { get; set; }
    public string Type { get; set; }
    public string SupplierName { get; set; }
    public DateOnly EntryDate { get; set; }
    public string ReviewUrl { get; set; }
}

File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/Souqcom/ProductsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductsRow } from "./ProductsRow";

export interface ProductsColumns {
    Id: Column<ProductsRow>;
    Name: Column<ProductsRow>;
    Discription: Column<ProductsRow>;
    Price: Column<ProductsRow>;
    CatidName: Column<ProductsRow>;
    Photo: Column<ProductsRow>;
    Type: Column<ProductsRow>;
    SupplierName: Column<ProductsRow>;
    EntryDate: Column<ProductsRow>;
    ReviewUrl: Column<ProductsRow>;
}

export class ProductsColumns extends ColumnsBase<ProductsRow> {
    static readonly columnsKey = 'Souqcom.Products';
    static readonly Fields = fieldsProxy<ProductsColumns>();
}